allow creating certs_tar for Katello app

sean797 opened the pull request:
For https://github.com/Katello/katello-installer/pull/628

ekohl commented on manifests/tar_create.pp:
```diff
@@ -10,11 +10,11 @@
 define certs::tar_create(
   $path               = $title,
-  $foreman_proxy_fqdn = $::certs::foreman_proxy_content::foreman_proxy_fqdn,
+  $fqdn               = $::certs::foreman_proxy_content::foreman_proxy_fqdn,
```
I wonder if this could overlap with the fqdn fact.

sean797 commented:
@ekohl I cleaned up the code, removed `foreman_proxy-content` class and added some tests. This is a breaking change for https://github.com/Katello/katello-installer/blob/master/bin/foreman-proxy-certs-generate#L65. I'll update https://github.com/Katello/katello-installer/pull/628 with all the required changes.

ekohl commented on manifests/generate_archive.pp:
```diff
@@ -0,0 +1,46 @@
+  String $server_fqdn          = undef,
+  Array[String] $server_cname  = $::certs::params::cname,
+  String[1] $certs_tar         = $::certs::params::certs_tar,
+  Boolean $foreman_application = false,
```
This is the general direction I'm going for but I'm not sure this is the granularity I'm aiming for. My goal is to split the node with candlepin and pulp into two separate nodes. Ideally qpid could run on a third. That means we'd bootstrap with candlepin as a node with the CA, then generate the certs for each additional node with the role they have. Do you think this makes sense?

sean797 replied:
Yes, and I agree its what we should aim for, well my end goal is being able to scale each component independently of each other.

Initially, I think we should try to make small(er) changes and iterate on them. Fewer chances of breakages and easier to review, we can extend the scope in other PRs later.

ekohl replied:
I've marked it as WIP for now so we don't accidentally merge it. I agree we should iterate on this but since it's a breaking change I'd like to keep it open until we're really happy with it.

sean797 commented:
I've made this backwards compatible as I want to protect `foreman-proxy-certs-generate`, If we were to make it use the new manifest then `--foreman-proxy-fqdn` arg would change to `--server-fqdn`.

ehelms commented on manifests/generate_archive.pp:
```diff
@@ -0,0 +1,46 @@
+# Prepare the certificates for the node from the parent node
```
Could you maintain backwards compatibility for the time being by having the foreman_proxy_content class call this class with parameters?

sean797 commented:
Thanks @ehelms, updated.

ekohl commented on .sync.yml:
```diff
@@ -8,7 +8,8 @@
 Rakefile:
   param_docs_pattern:
-    - manifests/foreman_proxy_content.pp
+    - manifests/foreman_proxy_content.pp 
```
Can you remove the trailing whitespace?

ekohl commented on manifests/tar_create.pp:
```diff
@@ -9,12 +9,12 @@
-  $path               = $title,
-  $foreman_proxy_fqdn = $::certs::foreman_proxy_content::foreman_proxy_fqdn,
+  $path    = $title,
+  $sub_dir = $::certs::foreman_proxy_content::foreman_proxy_fqdn,
```
Given we always pass this in I think we should remove the default value here. We're breaking the API anyway by changing the name.

sean797 commented:
Updated, I agree about `certs::tar_create` being an internal API ;)  

Thanks!
